larrecodnn/ImageMaker/SavePiMu_tool.cc

```diff
@@ -0,0 +1,150 @@
+
+    std::vector<art::Ptr<recob::Wire>> wires;
+    auto wireListHandle = e.getHandle<std::vector<recob::Wire>>("wclsdatasp:gauss");
+    if (wireListHandle) { art::fill_ptr_vector(wires, wireListHandle); }
+
+    // * MC truth information
+    std::vector<art::Ptr<simb::MCTruth>> mclist;
+    auto mctruthListHandle = e.getHandle<std::vector<simb::MCTruth>>("generator");
+    if (mctruthListHandle) art::fill_ptr_vector(mclist, mctruthListHandle);
+
+    unsigned short flag = 999;
+    if (!mclist.empty()) {
+      auto particle = mclist[0]->GetParticle(0);
+      //std::cout<<"pdg = "<<particle.PdgCode()<<std::endl;
+      if (std::abs(particle.PdgCode()) == 13) flag = 0;
+      if (std::abs(particle.PdgCode()) == 211) flag = 1;
+    }
+
+    if (!tracklist.empty()) {
+      auto trkend = tracklist[0]->End();
+      if (trkend.X() > -360 + 50 && trkend.X() < 360 - 50 && trkend.Y() > 50 &&
+          trkend.Y() < 610 - 50 && trkend.Z() > 50 && trkend.Z() < 710 - 50) {
+        //std::cout<<trkend.X()<<" "<<trkend.Y()<<" "<<trkend.Z()<<std::endl;
+        if (fmthm.isValid()) {
+          auto vhit = fmthm.at(0);
+          auto vmeta = fmthm.data(0);
+          int ihit = -1;
+          int maxindex = -1;
+          for (size_t i = 0; i < vhit.size(); ++i) {
+            if (vmeta[i]->Index() == std::numeric_limits<int>::max()) { continue; }
+            //std::cout<<i<<" "<<vmeta[i]->Index()<<std::endl;
+            if (vhit[i]->WireID().Plane == 2) {
+              if (int(vmeta[i]->Index()) > maxindex) {
+                maxindex = vmeta[i]->Index();
+                ihit = i;
+              }
+            }
+          }
+          if (ihit >= 0) {
+            //std::cout<<vhit[ihit]->WireID().toString()<<std::endl;
+            auto endwire = vhit[ihit]->WireID();
+            float endtime = vhit[ihit]->PeakTime();
+            float adc[50][50] = {{0.}};
+            for (auto& wire : wires) {
+              int channel = wire->Channel();
+              auto wireids = geom->ChannelToWire(channel);
+              //std::cout<<wireids[0].toString()<<std::endl;
+              if (wireids[0].Plane == 2 && wireids[0].TPC == endwire.TPC &&
+                  wireids[0].Wire >= endwire.Wire - 25 && wireids[0].Wire < endwire.Wire + 25) {
+                int idx = wireids[0].Wire - endwire.Wire + 25;
+                //std::cout<<"idx = "<<idx<<std::endl;
+                const recob::Wire::RegionsOfInterest_t& signalROI = wire->SignalROI();
+                int lasttick = 0;
+                for (const auto& range : signalROI.get_ranges()) {
+                  const auto& waveform = range.data();
+                  // ROI start time
+                  raw::TDCtick_t roiFirstBinTick = range.begin_index();
+                  for (int i = lasttick; i < roiFirstBinTick; ++i) {
+                    if (i >= int(endtime) - 25 && i < int(endtime) + 25) {
+                      adc[idx][i - int(endtime) - 25] = 0;
```

Comment:
A few more uses of hard-coded numbers (`25`).  Should this be a configuration parameter?